org.eclipse.lemminx/src/test/java/org/eclipse/lemminx/extensions/relaxng/xml/codeaction/missingChildElementCodeActionTest.java

```diff
@@ -0,0 +1,48 @@
+/*******************************************************************************
+* Copyright (c) 2022 Red Hat Inc. and others.
+* All rights reserved. This program and the accompanying materials
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v20.html
+*
+* SPDX-License-Identifier: EPL-2.0
+*
+* Contributors:
+*     Red Hat Inc. - initial API and implementation
+*******************************************************************************/
+package org.eclipse.lemminx.extensions.relaxng.xml.codeaction;
+
+import static org.eclipse.lemminx.XMLAssert.ca;
+import static org.eclipse.lemminx.XMLAssert.d;
+import static org.eclipse.lemminx.XMLAssert.te;
+import static org.eclipse.lemminx.XMLAssert.testCodeActionsFor;
+import static org.eclipse.lemminx.XMLAssert.testDiagnosticsFor;
+
+import org.eclipse.lemminx.AbstractCacheBasedTest;
+import org.eclipse.lemminx.extensions.relaxng.xml.validator.RelaxNGErrorCode;
+import org.eclipse.lsp4j.Diagnostic;
+import org.junit.jupiter.api.Test;
+
+/**
+ * Test for generating missing child elements for RELAX NG
+ *
+ */
+public class missingChildElementCodeActionTest extends AbstractCacheBasedTest {
+
+	@Test
+	public void incomplete_element_required_element_missing_simple() throws Exception {
+		String xml = "<?xml-model href=\"src/test/resources/relaxng/addressBook_v1.rng\" ?>\r\n" + //
+				"<addressBook>\r\n" + //
+				"  <card>\r\n" + //
+				"  </card>\r\n" + //
+				"</addressBook>";
+		Diagnostic d = d(2, 3, 2, 7, RelaxNGErrorCode.incomplete_element_required_element_missing);
+		testDiagnosticsFor(xml, d);
+		testCodeActionsFor(xml, d,
+				ca(d, te(2, 8, 2, 8, //
+						"\t\t<name></name>\r\n" + //
+								"\t\t<email></email>")),
+				ca(d, te(2, 8, 2, 8, //
+						"\t\t<name></name>\r\n" + //
+								"\t\t<email></email>")));
+	}
+}
```

Comment:
Test should looks like this:
 
```java
@Test
public void incomplete_element_required_element_missing_simple() throws Exception {
	String xml = "<?xml-model href=\"src/test/resources/relaxng/addressBook_v1.rng\" ?>\r\n" + //
			"<addressBook>\r\n" + //
			"	<card>\r\n" + //
			"	</card>\r\n" + //
			"</addressBook>";
	Diagnostic d = d(2, 2, 2, 6, RelaxNGErrorCode.incomplete_element_required_element_missing);
	testDiagnosticsFor(xml, d);
	testCodeActionsFor(xml, d,
			ca(d, te(2, 7, 2, 7, //
					"\r\n" + //
					"\t\t<name></name>\r\n" + //
							"\t\t<email></email>")),
			ca(d, te(2, 7, 2, 7, //
					"\r\n" + //
					"\t\t<name></name>\r\n" + //
							"\t\t<email></email>")));
}
```

In otherwise:

 * you should use tabulation to indent XML (you are using spaces)
 * it misses the `\r\n` before generation of XML element

The test fails because there is a bug with isOptional in CMRelaxNGElementDeclaration.